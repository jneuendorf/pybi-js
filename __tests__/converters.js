const ascii = require('../src/ascii')
const bin = require('../src/bin')
const bool = require('../src/bool')
const bytearray = require('../src/bytearray')
const bytes = require('../src/bytes')
const chr = require('../src/chr')
const dict = require('../src/dict')
const float = require('../src/float')
const frozenset = require('../src/frozenset')
const hex = require('../src/hex')
const int = require('../src/int')
const list = require('../src/list')
const object = require('../src/object')
const oct = require('../src/oct')
const ord = require('../src/ord')
const set = require('../src/set')
const str = require('../src/str')
const tuple = require('../src/tuple')

const {ValueError} = require('../src/_errors')

const {createTestCase} = require('./_utils')


test('ascii', () => {
    expect(ascii(undefined)).toBe('undefined')
    expect(ascii(null)).toBe('null')
    expect(ascii('ipsum áá éé lore')).toBe('ipsum \\u00e1\\u00e1 \\u00e9\\u00e9 lore')
    expect(ascii({
        __repr__() {
            return 'Göt you?'
        }
    })).toBe('G\\u00f6t you?')
})


describe('bin', () => {
    createTestCase('converters', 'bin', bin, {testName: 'without __index__'})

    test('with __index__', () => {
        expect(bin({
            __index__() {
                return 42
            }
        })).toBe(bin(42))
    })
})


describe('bool', () => {
    createTestCase('converters', 'bool', bool, {testName: 'without __bool__'})

    test('with __bool__', () => {
        expect(bool({
            __bool__() {
                return true
            }
        })).toBe(true)
        expect(bool({
            __bool__() {
                return false
            }
        })).toBe(false)
        expect(() => bool({
            __bool__() {
                return 1
            }
        })).toThrow(TypeError)
    })
})


describe('bytearray', () => {
    createTestCase('converters', 'bytearray', bytearray, {
        testName: 'without buffer interface objects',
        logIndices: true,
        deserializer: obj => Uint8Array.from(obj),
    })

    test('with buffer', () => {
        expect(
            bytearray(Buffer.from('aä', 'utf8'))
        ).toEqual(Uint8Array.from([97, 195, 164]))
    })
})


describe('a', () => {
    test('b', () => {

    })

    test('c', () => {

    })
})
