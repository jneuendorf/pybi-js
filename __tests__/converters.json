{
    "bin": [
        [42],
        [3],
        [-10],
        ["what?"]
    ],
    "bool": [
        [],
        [null],
        [0],
        [1],
        [-1],
        [""],
        ["asdf"],
        [[]],
        [[1, 2]],
        [{}],
        [{"a": 1}],
        [true],
        [false]
    ],
    "bytearray": [
        [],
        ["asdf"],
        ["äsdf", "utf8"],
        [10],
        [[1, 5, 10]],
        [["a", "b"]],
        [[100, 500, 1000]]
    ],
    "bytes": [
        [],
        ["asdf"],
        ["äsdf", "utf8"],
        [10],
        [[1, 5, 10]],
        [["a", "b"]],
        [[100, 500, 1000]]
    ],
    "chr": [
        [0],
        [97],
        [8364],
        [1114111],
        [1114112]
    ]
}
