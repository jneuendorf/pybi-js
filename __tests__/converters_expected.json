{
  "bin": [
    "0b101010",
    "0b11",
    "-0b1010",
    {
      "__error__": {
        "type": "TypeError",
        "message": "'str' object cannot be interpreted as an integer"
      }
    }
  ],
  "bool": [
    false,
    false,
    false,
    true,
    true,
    false,
    true,
    false,
    true,
    false,
    true,
    true,
    false
  ],
  "bytearray": [
    [],
    {
      "__error__": {
        "type": "TypeError",
        "message": "string argument without an encoding"
      }
    },
    [
      195,
      164,
      115,
      100,
      102
    ],
    [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0
    ],
    [
      1,
      5,
      10
    ],
    {
      "__error__": {
        "type": "TypeError",
        "message": "'str' object cannot be interpreted as an integer"
      }
    },
    {
      "__error__": {
        "type": "ValueError",
        "message": "byte must be in range(0, 256)"
      }
    }
  ],
  "bytes": [
    [],
    {
      "__error__": {
        "type": "TypeError",
        "message": "string argument without an encoding"
      }
    },
    [
      195,
      164,
      115,
      100,
      102
    ],
    [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0
    ],
    [
      1,
      5,
      10
    ],
    {
      "__error__": {
        "type": "TypeError",
        "message": "'str' object cannot be interpreted as an integer"
      }
    },
    {
      "__error__": {
        "type": "ValueError",
        "message": "bytes must be in range(0, 256)"
      }
    }
  ],
  "chr": [
    "\u0000",
    "a",
    "\u20ac",
    "\udbff\udfff",
    {
      "__error__": {
        "type": "ValueError",
        "message": "chr() arg not in range(0x110000)"
      }
    }
  ]
}